### PR TITLE
Use riff-raff.yaml instead of deploy.json

### PR DESCRIPTION
**Pull request by AWare**
Updating to produce the directory structure required by riff-raff.yaml.

Ready for testing by running 
`yarn add guardian/node-riffraff-artefact#37919908e3eabcf2c4afb66cf328547836bf35b6 --dev`
or `npm install --save-dev guardian/node-riffraff-artefact#37919908e3eabcf2c4afb66cf328547836bf35b6` 

todo:
- [x] new version number
- [x] exit(1) if deploy.json used instead of riff-raff.yaml

### CODE REVIEW

**Review by akash1810 — COMMENTED** [review]
should we try and parse the `riffraffFile` and throw an error if its not recognised? Riff Raff will fail to deploy in this scenario anyway though...

File: src/lib/util.js [hunk 1]
```diff
@@ -96,7 +76,6 @@ module.exports = {
     log,
     createDir,
     createZip,
-    createTar,
```
Comment (piuccio) [inline src/lib/util.js]: I think `facia-tool` is using `tgz`

Comment (AWare) [reply src/lib/util.js]: Yeah, I realised a bit too late that "isLambda":false was a useless search.

**Comment by AWare** [comment]
@akash1810 riffraff already has https://riffraff.gutools.co.uk/configuration/validation
would we just check for valid yaml?

**Comment by akash1810** [comment]
> would we just check for valid yaml?

👍

File: src/riffraff-artefact.js [hunk 2]
```diff
@@ -187,12 +164,24 @@ function determineAction() {
     (SETTINGS.env !== "dev" && SETTINGS.uploadArtefact) ? buildAndDeployArtefact() : buildArtefact();
 }
 
+function validateYaml(){
```
Comment (akash1810) [inline src/riffraff-artefact.js]: minor: `isValidYaml` might be more expressive

File: src/settings.js [hunk 3]
```diff
@@ -18,17 +18,18 @@ log("Reading configuration from " + ROOT + "/package.json");
...
 
 const uploadArtefact = packageJson.uploadArtefact === undefined ? true : packageJson.uploadArtefact;
 
 let SETTINGS = {
     rootDir: ROOT,
-    artefactsFilename: "artifacts.zip",
     packageName: packageJson.name,
     manifestProjectName: projectName || packageJson.name,
     cloudformation: (cf == undefined) ? "cloudformation.json" : cf,
+    cloudformationAppName: (cloudformationAppName == undefined) ? packageJson.name + "-cloudformation" : cloudformationAppName,
```
Comment (akash1810) [inline src/settings.js]: could we add a description of `cloudformationAppName` to the readme?

Comment (AWare) [reply src/settings.js]: Yeah, I'd like to put a bit more in there about rifffraff.yaml too

**Review by akash1810 — APPROVED** [review]

**Pull request merged**